BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -408,25 +406,31 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
         self.updateDeadbeatHeight()
         self.lastUpdated = Date()
         self.updateLastUpdatedLabel()
-        if self.goals.count == 0 {
-            self.noGoalsLabel.isHidden = false
-            self.collectionView?.isHidden = true
-        } else {
-            self.noGoalsLabel.isHidden = true
-            self.collectionView?.isHidden = false
-        }
-        let searchItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(self.searchButtonPressed))
-        self.navigationItem.leftBarButtonItem = searchItem
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(self.searchButtonPressed))
     }
 
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         return ServiceLocator.versionManager.lastChckedUpdateState() == .UpdateRequired ? 0 : self.filteredGoals.count
     }
     
     func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
+        [filteredGoals.isEmpty, goals.isEmpty].contains(true) ? 0 : 1
```

Comment:
```suggestion
        filteredGoals.isEmpty ? 0 : 1
```
If `goals` is empty then filtered goals will always be, right?

Reply:
Yes, if `goals` is empty, then any subset of it would also be empty, including the one known as `filteredGoals`.

I see what you mean: we are always using the combination of the filter applied to the goals and thus the set in `filteredGoals`. When the searchBar.text the filtered is considered off and all goals are in `filteredGoals`.

Reply:
Perhaps these names are unclear. Something like `allGoals` and `visibleGoals` might have been a better choice?